src/vellum/workflows/expressions/contains.py

```
@@ -23,6 +24,9 @@ def resolve(self, state: "BaseState") -> bool:
         # Support any type that implements the in operator
         # https://app.shortcut.com/vellum/story/4658
         lhs = resolve_value(self._lhs, state)
+        # assumes that lack of is also false
+        if lhs is UNDEF:
+            return False
         if not isinstance(lhs, (list, tuple, set, dict, str)):
             raise ValueError(f"Expected a LHS that supported contains, got: {lhs.__class__.__name__}")
```

[q] Was this the error that was being throws? Wondering if we should just return false here

---

[oos] Actually, I think we still want to raise an error here for other non-undefined types but make it rise up to a user error instead of internal error. I'll patch that up now

---

Heyo! Sorry for the late reply, but the error is that the conditional was against an error that is undefined, due to there being no error inside of the try/catch and so the type error below was being thrown

---

If we don't want this here, we can probably make it so Error is always at least None?

---

Yea I think we could keep this here, followed up with the other changes I had in mind